app/services/foreman/plugin.rb

```diff
@@ -83,6 +83,12 @@ def find(id)
         registered_plugins[id.to_sym]
       end
 
+      # Evaluates additional setup steps after initial registration
+      # useful when you need to define methods later e.g. from config.to_prepare
+      def setup(id, &block)
```

Comment:
Using class directly has small benefits like early syntax check and editor navigation. I introduced `in_to_prepare` plugin helper that registers callback and makes sure it's run later in `to_prepare` which is called after reload. It's also handy for new allowed helpers and variables since after reload they need to inject arguments into newly loaded Renderer.